tests/core/vm/test_computation.py

```diff
@@ -0,0 +1,184 @@
+import pytest
+
+from evm.vm import (
+    Computation,
+    Message,
+)
+
+from eth_utils import (
+    to_canonical_address,
+)
+
+from evm.exceptions import (
+    ValidationError,
+)
+
+CANONICAL_ADDRESS_1 = to_canonical_address("0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6")
+CANONICAL_ADDRESS_2 = to_canonical_address("0xcd1722f3947def4cf144679da39c4c32bdc35681")
+
+
+@pytest.fixture
+def message():
+    message = Message(
+        origin=to_canonical_address(CANONICAL_ADDRESS_2),
+        to=CANONICAL_ADDRESS_1,
+        sender=to_canonical_address(CANONICAL_ADDRESS_2),
+        value=100,
+        data=b"0x",
+        code=b"0x",
+        gas=100,
+        gas_price=1,
+    )
+    return message
+
+
+@pytest.fixture
+def computation(message):
+    computation = Computation(
+        vm=1,
+        message=message
+    )
+    return computation
+
+
+@pytest.fixture
+def create_error():
+    def computation_error(computation):
+        computation.gas_meter.gas_remaining = 0
```

Comment:
I don't think this is a good approach.  Things I dislike are that 1) it directly mutates the `gas_meter` to produce an OOG error and 2) the error is not *authentic*.

It's pretty trivial to generate a VM error using some simple invalid bytecode.  The bytecode `b'\x50` which is a single `POP` instruction should throw an error as long as the stack is empty.  I'm not exactly sure how the best way to apply this to these tests is yet but leaving this here to at least surface the problem.